live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/auth/AuthenticationService.java

```diff
@@ -0,0 +1,177 @@
+import org.apache.camel.Exchange;
+import org.springframework.boot.autoconfigure.security.oauth2.client.OAuth2ClientProperties;
+import org.springframework.scheduling.TaskScheduler;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContext;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.oauth2.client.authentication.OAuth2AuthenticationToken;
+import org.springframework.security.oauth2.client.authentication.OAuth2LoginAuthenticationToken;
+import org.springframework.security.oauth2.core.OAuth2AccessToken;
+import org.springframework.security.oauth2.core.user.DefaultOAuth2User;
+import org.springframework.security.oauth2.core.user.OAuth2User;
+import org.springframework.security.oauth2.core.user.OAuth2UserAuthority;
+import org.springframework.security.scheduling.DelegatingSecurityContextTaskScheduler;
+import org.springframework.stereotype.Service;
+
+import org.alfresco.hxi_connector.live_ingester.adapters.config.IntegrationProperties;
+import org.alfresco.hxi_connector.live_ingester.adapters.config.properties.Authorization;
+import org.alfresco.hxi_connector.live_ingester.domain.exception.LiveIngesterRuntimeException;
+
+@Service
+@RequiredArgsConstructor
+public class AuthenticationService
+{
+    private static final String CLIENT_REGISTRATION_ID = "hyland-experience-auth";
+    private static final String APP_NAME_ATTRIBUTE_KEY = "applicationName";
+    private static final String SERVICE_USER_ATTRIBUTE_KEY = "serviceUser";
+    private static final String ENVIRONMENT_KEY_ATTRIBUTE_KEY = "hxAiEnvironmentKey";
+    private static final String ENVIRONMENT_KEY_HEADER = "hxai-environment";
+    private static final int AUTH_SCHEDULE_DELAY_MINUTES = 55;
+    private static final int WAIT_FOR_PAUSE_TIME_MILLIS = 100;
+
+    private final OAuth2ClientProperties oAuth2ClientProperties;
+    private final IntegrationProperties integrationProperties;
+    private final AuthenticationManager authenticationManager;
+    private final TaskScheduler taskScheduler;
+    private final CamelContext camelContext;
+
+    // Temporary disabled
+    // @PostConstruct
+    public void authenticationSchedule()
+    {
+        SecurityContext securityContext = SecurityContextHolder.getContext();
+        DelegatingSecurityContextTaskScheduler delegatingTaskScheduler = new DelegatingSecurityContextTaskScheduler(taskScheduler, securityContext);
+        Runnable authenticationTask = () -> {
+            waitFor(camelContext::isStarted);
+            authenticate();
+        };
+        delegatingTaskScheduler.scheduleWithFixedDelay(authenticationTask, Duration.ofMinutes(AUTH_SCHEDULE_DELAY_MINUTES));
+    }
+
+    public void authenticate()
+    {
+        authenticate(false);
+    }
+
+    public void authenticate(boolean forceAuthentication)
```

Comment:
This doesn't ever seem to be set to anything but `false`.

Reply:
This is a leftover after testing 🤔  You are right, that this is not required at this moment - I will remove it